### PR TITLE
Misc options rendering adjustments

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

Code clarity and content vs presentation. See each commit for details.

# Context

Follow-up from #8931.

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: doc/manual/generate-manpage.nix [hunk 1]
```diff
@@ -107,8 +109,12 @@ let
 
                 ${option.description}
             '';
-          categories = sort lessThan (unique (map (cmd: cmd.category) (attrValues allOptions)));
-        in concatStrings (map showCategory categories);
+          categories = mapAttrs
+            (_: listToAttrs)
```
Comment (fricklerhandwerk) [inline doc/manual/generate-manpage.nix:113]: This confuses me. Why does the function that mapAttrs applies return listToAttrs?

**Review by fricklerhandwerk — APPROVED** [review]
Except for that one line I don't understand, this is a very sensible change.

File: doc/manual/generate-manpage.nix (hunk 1 above)
Comment (Ericson2314) [reply doc/manual/generate-manpage.nix:113]: The undoes the `attrsToList` further below.

We have:

- `allOptions : AttrSet<Option>`
- `attrsToList allOptions : List<{ name: String, value: Option }>`
- `groupBy ... : AttrSet<List<{ name: String, value: Option }>>`
- `mapAtrs ... : AttrSet<AttrSet<Option>>`

Comment (fricklerhandwerk) [reply doc/manual/generate-manpage.nix:113]: Ah yes, this is subtle. Maybe
```suggestion
            (_: listToAttrs) # undo `attrsToList allOptions` for each group
```

Comment (Ericson2314) [reply doc/manual/generate-manpage.nix:113]:
```suggestion
            # Convert each group from a list of key-value pairs back to an attrset
            (_: listToAttrs)
```

**Pull request merged**